rustls/src/error.rs

```diff
@@ -1166,6 +1166,9 @@ pub enum ApiMisuse {
 
     /// QUIC attempted with a configuration that does not support TLS1.3.
     QuicApiRequiresTls13Support,
+
+    /// QUIC attempted with a configuration that does not support a ciphersuite that supports QUIC.
+    QuicApiRequiresAtLeastOneCipherSuiteSupportingQuic,
```

Comment:
Can we get this a bit more concise? `NoQuicCompatibleCipherSuites`?

Maybe the `Api` part of the `QuicApi` prefix also doesn't add much value?